sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -280,6 +280,29 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
   }
 
   /**
+   * Extracts the [[StructField]] with the given name recursively.
+   *
+   * @throws IllegalArgumentException if the parent field's type is not StructType
+   */
+  def getFieldRecursively(name: String): StructField = {
```

Comment:
Isn't this Parquet-specific problem? I wonder adding this method is appropriate.

Also, I am not too sure if it is appropriate to mark nested fields by modifying field names with a character.